linux/cmd/df/df.go

```diff
@@ -1,5 +1,68 @@
 package df
 
-func Df() {
+import (
+	"os/exec"
+	"strconv"
+	"strings"
+)
 
+type DiskUsage struct {
+	Filesystem string `json:"filesystem"`
+	Blocks     int    `json:"512-blocks"`
+	Used       int    `json:"used"`
+	Available  int    `json:"available"`
+	Capacity   string `json:"capacity"`
+	Iused      int    `json:"iused"`
+	Ifree      int    `json:"ifree"`
+	IusedPerc  string `json:"%iused"`
+	MountedOn  string `json:"mounted_on"`
+}
+
+func Df() ([]DiskUsage, error) {
+	cmd := exec.Command("df", "-i")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return nil, err
+	}
+
+	lines := strings.Split(string(output), "\n")
+	diskUsages := make([]DiskUsage, 0, len(lines)-1)
+
+	for _, line := range lines[1:] {
+		if line == "" {
+			continue
+		}
+
+		fields := strings.Fields(line)
+		if fields[0] == "map" {
+			// Filesystem is "map auto_home"
+			fields[1] = "map " + fields[1]
+			fields = fields[1:]
+		}
```

Comment:
It seems too sporadic... There should typically be single spaces in filesystem names, so could we identify information from other categories by noticing that the spacing between characters is more than two?

Reply:
In the example below, you can see that the number of spaces between data of `Used` and `Available` is 1.
It means that there should be the case when there is a single space between `Filesystem` and `512-blocks`.
```
Filesystem     512-blocks      Used Available Capacity iused      ifree %iused  Mounted on
/dev/disk3s1s1  478724992  20844296 225635920     9%  407401 1128179600    0%   /
devfs                 711       711         0   100%    1230          0  100%   /dev
```

Reply:
Okay, I'll let it go as an occasional case.